src/Lykke.Service.Stellar.Api.Services/Horizon/HorizonService.cs

```diff
@@ -49,12 +49,18 @@ public async Task<string> SubmitTransactionAsync(string signedTx)
             var acc = await _server.Accounts.Account(transaction.SourceAccount.AccountId);
             var tx = await _server.SubmitTransaction(transaction);
 
-
-            if (tx == null || string.IsNullOrEmpty(tx.Hash))
+            if (string.IsNullOrEmpty(tx?.Hash))
             {
+                if (!tx.Result.IsSuccess)
```

Comment:
It looks like NRE possible here. Shouldn't we throw HorizonApiException in the case when tx is null and if not, then check IsSuccess?